### ISSUE TITLE
right command key invokes context menu on Mac OS

### ISSUE THREAD

**Issue by aol-nnov** [ONLYOFFICE/DesktopEditors#305]
**Do you want to request a *feature* or report a *bug*?**

**What is the current behavior?**
if right command (⌘) key is pressed (alone or in key combo), context menu appears

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
context menu should no be shown (at least, this behaviour should be customisable)

**Which versions of DesktopEditors, and which OS are affected by this issue? Did this work in previous versions of DesktopEditors?**

Onlyoffice version 5.4.2 (367), also reproducible in browser (cloud) version 5.4.1 (which is run locally from official docker container
host os: Mac OS Catalina 10.15.2 (19C57), also reproducible in High Sierra
browser: Google chrome Version 79.0.3945.117 (Official Build) (64-bit)

**Comment by K0R0L** [comment]
This bug will be fixed in version 5.5.

**Comment by ShockwaveNN** [comment]
Leave this issue opened, since DesktopEditors 5.5 not released yet

**Comment by ShockwaveNN** [comment]
DesktopEditors 5.5.1 released. This issue should be fixed.

**Comment by chriswep** [comment]
@ShockwaveNN i'm on 6.2 (492) and the cmd key still opens the context menu (and did so in all the last 6.* versions i used). unfortunately this makes the software almost unusable.

**Comment by ShockwaveNN** [comment]
@chriswep Ok, thanks I reopened this issue

@DenisDeeSign Please take a look

**Comment by aol-nnov** [comment]
I can confirm, this issue was fixed for some time and after a few releases it stroke back

**Comment by chrisknepper** [comment]
For me this occurs with the left command key but not the right in macOS Big Sur.
Document Editor 6.2.2
If I use right command key I can copy and paste and no context menu appears
If I use left command key, the document editor context menu appears and does not copy nor paste

**Comment by chriswep** [comment]
I can confirm that the right command key does indeed work as intended on my desktop mac with usb keyboard. I now also tested the issue on my MacBook with internal keyboard and funny enough - there the left command key works as intended however the right command key opens the context menu. I'm using UK keyboard layout on both. But even if i change the keyboard layout, the issue remains the same. @ShockwaveNN

**Comment by ShockwaveNN** [comment]
@chriswep Sorry, I don't have mac at my disposal, so we should wait for @DenisDeeSign to return from vacation to take a look at this issue once again

**Comment by chrisknepper** [comment]
@chriswep @ShockwaveNN 

I am pretty sure I understand when and why this happens now.

In macOS, there is a system option to switch keyboard modifier keys around. Typical use case is to swap command and option keys because in Apple keyboard layout, the command key is immediately left of the spacebar, but on non-Apple keyboards, the alt key is immediately left of the spacebar. Without changing this preference, using keyboard shortcuts involving the command (now Windows/"super" key) and/or option key (now alt key) can field unnatural if one is used to using a real Apple keyboard.

I indeed had these keys swapped via System Preferences > Keyboard > Modifier Keys... (see first attached screenshot) and when I reset them to default, the ONLYOFFICE document editors did not show this "bug."

I am putting "bug" in quotes because I actually feel this is a poor implementation on Apple's side. When a user swaps a key to act as another key at the system level, the system should then send that other key's `keyCode` in JavaScript contexts.

To work around this, I am using another open source program, https://github.com/pqrs-org/Karabiner-Elements which allows much more complex manipulation and key remapping. With a simple rule (see second attached screenshot), I can now have the best of both worlds: On my non-Apple keyboard, the left alt key acts as the command key, and the left Windows key acts as the option key, and ONLYOFFICE no longer shows the context menu when it should not (for example, when using command+c to copy text).

tl;dr this looks like a problem exclusive to non-Apple or external keyboards combined with the modifier keys option in macOS.

As a potential solution on the ONLYOFFICE side, perhaps the `metaKey` event property can be used? https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/metaKey

Hopefully this helps someone else out there experiencing this!

<img width="780" alt="Screen Shot 2021-06-06 at 3 34 16 PM" src="https://user-images.githubusercontent.com/4976096/120940167-eebd6000-c6e9-11eb-88b1-f20f66fb7f46.png">

<img width="1112" alt="Screen Shot 2021-06-06 at 5 08 48 PM" src="https://user-images.githubusercontent.com/4976096/120940157-e2d19e00-c6e9-11eb-9d3d-70ac229d4b91.png">

**Comment by chriswep** [comment]
@chrisknepper i also suspected something about that macos-swapping since i have that enabled on my desktop. however that doesn't explain the same issue on the right cmd key of my internal macbook keyboard (no swapping enabled). also, something deeper seems to be going on, since swapping alt and cmd (in macOS settings) makes OnlyOffice detect cmd/win=alt, ctrl=cmd and alt=ctrl.

**Comment by ShockwaveNN** [comment]
We got somehow related issue that was resolved in v6.3.0 via
https://github.com/ONLYOFFICE/sdkjs/commit/6a1df3d17c941378c6d15f01b262c035fee0ff1c

We'll release DesktopEditors of v6.3.0 soon and maybe your cases will be also resolved, but I'm not 100% sure since I, personally, have no access to mac hardware

**Comment by ShockwaveNN** [comment]
DocumentServer and DesktopEditors v6.3.1 is released, but i'm not sure if this issue is resolved, since I'm personally have no Mac at my disposal

**Comment by chriswep** [comment]
@ShockwaveNN did a (very) quick first test and cmd does not open the context menu anymore, so the update seems to have fixed things, at least on my side

**Comment by ShockwaveNN** [comment]
@chriswep In that case I'll close this issue, but feel free to comment here or open new issue if some troubles arise

**Comment by aol-nnov** [comment]
I can confirm, that right cmd key does not invoke context menu in DesktopEditors v6.3.1 and latest docker image of Community DocumentServer.

please do not break it one more time ;) it was a very annoying bug..

**Comment by ShockwaveNN** [comment]
@aol-nnov Thank you for your responce, we will try to )

**Comment by zilbaouf93** [comment]
@ShockwaveNN Hey,
I'm having this exact same issue on v8.2.2. I tried using the command, option or control keys but without success, and I don't have any key remapping.

**Comment by Gingetsu** [comment]
Hello, @zilbaouf93!
I can't reproduce your problem.
Create a new issue and describe it in more detail, please.
Thank you for report!
